kubernetes/resource_kubernetes_deployment.go

```diff
@@ -385,6 +385,8 @@ func waitForDeploymentReplicasFunc(conn *kubernetes.Clientset, ns, name string)
 			if dply.Status.AvailableReplicas < dply.Status.UpdatedReplicas {
 				return resource.RetryableError(fmt.Errorf("Waiting for rollout to finish: %d of %d updated replicas are available...", dply.Status.AvailableReplicas, dply.Status.UpdatedReplicas))
 			}
+		} else if dply.Status.ObservedGeneration == 0 {
```

Comment:
Can `dply.Generation` ever be `0` and thus also satisfy the first branch of the condition?

Reply:
We haven’t seen it so far. 

It’s either 1 and 1 or 1 and 0 

We see this when we deploy tiller on a fresh AKS using k8s provider. The helm provider follows up with an error that tiller isn’t installed once in a while.